CalibTracker/SiStripCommon/plugins/ShallowEventDataProducer.cc

```diff
@@ -3,15 +3,21 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
+// #include <bitset>
+
 ShallowEventDataProducer::ShallowEventDataProducer(const edm::ParameterSet& iConfig) {
   produces <unsigned int> ( "run"      );
   produces <unsigned int> ( "event"    );
   produces <unsigned int> ( "bx"       );
   produces <unsigned int> ( "lumi"     );
   produces <float>        ( "instLumi" );
   produces <float>        ( "PU"       );
+  #ifdef CALIBTreeDEV
   produces <std::vector<bool> > ( "TrigTech" );
   produces <std::vector<bool> > ( "TrigPh" );
+  // produces <std::bitset<64> > ( "TrigTechBitSet" );
+  // produces <std::bitset<128> > ( "TrigPhBitSet" );
```

Comment:
@mmusich , the first idea was to use bitset instead of a vector of boolean in order to save space, but it is not sure if it would be the case, so I will continue to investigate. For the moment the commented line can be canceled also because the trigger boolean vectors are marked as _unused_ in the survey.

Reply:
I quickly add a look. I'm not wrong you wil save space by using std::bitset<X>.
A vector of 64 bool use 104 bytes:
 sizeof(std::vector<bool>) + sizeof(bool) * 64
while by construction std::biset(64) use 8 bytes